src/base/types/derivedmodels/GeckoModel.jl

```diff
@@ -0,0 +1,423 @@
+"""
+    mutable struct GeckoModel <: MetabolicModel
+
+A model that incorporates enzyme capacity and kinetic constraints via the GECKO
+formulation. See `Sánchez, Benjamín J., et al. "Improving the phenotype
+predictions of a yeast genome‐scale metabolic model by incorporating enzymatic
+constraints." Molecular systems biology, 2017.` for implementation details.
+
+Note, since the model uses irreversible reactions internally, `"§FOR"` (for the
+forward direction) and `"§REV"` (for the reverse direction) is appended to each
+reaction internally. Hence, `"§"` is reserved for internal use as a delimiter,
+no reaction id should contain this character. 
+
+To actually run GECKO, call [`flux_balance_analysis`](@ref) on a `GeckoModel`.
+
+# Fields
+```
+reaction_ids::Vector{String}
+irrev_reaction_ids::Vector{String}
+metabolites::Vector{String}
+gene_ids::Vector{String}
+c::SparseVec
+S::SparseMat
+b::SparseVec
+xl::SparseVec
+xu::SparseVec
+C::SparseMat
+cl::Vector{Float64}
+cu::Vector{Float64}
+```
+"""
+mutable struct GeckoModel <: MetabolicModel
+    reaction_ids::Vector{String}
+    irrev_reaction_ids::Vector{String}
+    metabolites::Vector{String}
+    gene_ids::Vector{String}
+
+    # gecko  
+    c::SparseVec
+    S::SparseMat
+    b::SparseVec
+    xl::SparseVec
+    xu::SparseVec
+
+    # enzyme capacity constraints
+    C::SparseMat
+    cl::Vector{Float64}
+    cu::Vector{Float64}
+end
+
+"""
+    stoichiometry(model::GeckoModel)
+
+Return stoichiometry matrix that includes enzymes as metabolites.
+"""
+stoichiometry(model::GeckoModel) = model.S
+
+"""
+    balance(model::GeckoModel)
+
+Return stoichiometric balance.
+"""
+balance(model::GeckoModel) = model.b
+
+"""
+    objective(model::GeckoModel)
+
+Return objective of `model`.
+"""
+objective(model::GeckoModel) = model.c
+
+"""
+    reactions(model::GeckoModel)
+
+Returns the reversible reactions in `model`. For 
+the irreversible reactions, use [`irreversible_reactions`][@ref].
+"""
+reactions(model::GeckoModel) = model.reaction_ids
+
+"""
+    n_reactions(model::GeckoModel)
+
+Returns the number of reactions in the model.
+"""
+n_reactions(model::GeckoModel) = length(model.reaction_ids)
+
+"""
+    irreversible_reactions(model::GeckoModel)
+
+Returns the irreversible reactions in `model`.
+"""
+irreversible_reactions(model::GeckoModel) = model.irrev_reaction_ids
```

Comment:
`irreversible_reactions` are precisely `reactions` now -- you can put anything you want into reactions. The "good old reactions" should be contained in fluxes().